components/omega/src/infra/Broadcast.h

```diff
@@ -0,0 +1,98 @@
+#ifndef OMEGA_BROADCAST_H
+#define OMEGA_BROADCAST_H
+//===-- infra/Broadcast.h - Broadcasting values --*- C++ -*-===//
+//
+/// \file
+/// \brief Defines MPI broadcasting functions
+///
+/// This header defines functions to broadcast values from one MPI rank
+/// to another.
+//
+//===----------------------------------------------------------------------===//
+
+#include "DataTypes.h"
+#include "MachEnv.h"
+#include "mpi.h"
+
+namespace OMEGA {
+
+// blocking broadcast scalar
+int Broadcast(I4 &value, const MachEnv *InEnv = MachEnv::getDefaultEnv(),
```

Comment:
One big change here. We allow the MachEnv to define a MasterTask other than 0, so in all these interfaces, we want to set the argument RankBcast to an invalid value and extract MasterTask from the MachEnv.

Reply:
Thanks a lot for pointing this. Now, if user does not specify RankBcast value(default value is set to an invalid number of "-1"), the module get it using getMasterTask().